test/integrations/destinations/ga4/dataDelivery/business.ts

```diff
@@ -0,0 +1,276 @@
+import { ProxyMetdata } from '../../../../../src/types';
+import { generateProxyV1Payload } from '../../../testUtils';
+import { JSON_MIME_TYPE } from '../../../../../src/v0/util/constant';
+
+const headers = {
+  HOST: 'www.google-analytics.com',
+  'Content-Type': JSON_MIME_TYPE,
+};
+
+const params = {
+  api_secret: 'dymmyApiSecret',
+};
+
+const validRequest = {
+  events: [
+    {
+      name: 'sign_up',
+      params: {
+        method: 'google',
+        engagement_time_msec: 1,
+      },
+    },
+  ],
+  user_id: 'dummyUserId',
+  client_id: 'dummyClientId',
+  non_personalized_ads: true,
+};
+
+const invalidEventNameRequest = {
+  events: [
+    {
+      name: 'campaign@details',
+      params: {
+        term: 'summer+travel',
+        medium: 'cpc',
+        source: 'google',
+        content: 'logo link',
+        campaign: 'Summer_fun',
+        campaign_id: 'google_1234',
+        engagement_time_msec: 1,
+      },
+    },
+  ],
+  user_id: 'dummyUserId',
+  client_id: 'dummyClientId',
+  non_personalized_ads: true,
+};
+
+const invalidParameterValueRequest = {
+  events: [
+    {
+      name: 'add_to_cart',
+      params: {
+        currency: 'USD',
+        value: 7.77,
+        engagement_time_msec: 1,
+        items: [
+          {
+            item_id: '507f1f77bcf86cd799439011',
+            item_name: 'Monopoly: 3rd Edition',
+            coupon: 'SUMMER_FUN',
+            item_category: 'Apparel',
+            item_brand: 'Google',
+            item_variant: 'green',
+            price: '$19',
+            quantity: 2,
+            affiliation: 'Google Merchandise Store',
+            currency: 'USD',
+            item_list_id: 'related_products',
+            item_list_name: 'Related Products',
+            location_id: 'L_12345',
+          },
+        ],
+      },
+    },
+  ],
+  user_id: 'dummyUserId',
+  client_id: 'dummyClientId',
+  non_personalized_ads: true,
+};
+
+const proxyMetdata: ProxyMetdata = {
+  jobId: 1,
+  attemptNum: 1,
+  userId: 'dummyUserId',
+  sourceId: 'dummySourceId',
+  destinationId: 'dummyDestinationId',
+  workspaceId: 'dummyWorkspaceId',
+  secret: {},
+  dontBatch: false,
+};
+
+const metadataArray = [proxyMetdata];
+
+export const testScenariosForV1API = [
```

Comment:
Use `ProxyV1TestData[]` for test cases type

Reply:
addressed